src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+        {
+            LOG_LAST_ERROR();
+            // If SystemParametersInfoW fails, which it shouldn't, fall back to
+            // Windows' default value.
+            _rowsToScroll = 3;
+        }
+    }
+
+    // Method Description:
+    // - Given a copy-able selection, get the selected text from the buffer and send it to the
+    //     Windows Clipboard (CascadiaWin32:main.cpp).
+    // - CopyOnSelect does NOT clear the selection
+    // Arguments:
+    // - singleLine: collapse all of the text to one line
+    // - formats: which formats to copy (defined by action's CopyFormatting arg). nullptr
+    //             if we should defer which formats are copied to the global setting
+    bool ControlInteractivity::CopySelectionToClipboard(bool singleLine,
+                                                        const Windows::Foundation::IReference<CopyFormat>& formats)
+    {
+        if (_core)
+        {
+            // Note to future self: This should return false if there's no
+            // selection to copy. If there's no selection, returning false will
+            // indicate that the actions that triggered this should _not_ be
+            // marked as handled, so ctrl+c without a selection can still send
+            // ^C
+
+            // Mark the current selection as copied
+            _selectionNeedsToBeCopied = false;
+
+            return _core->CopySelectionToClipboard(singleLine, formats);
+        }
+
+        return false;
+    }
+
+    // Method Description:
+    // - Initiate a paste operation.
+    void ControlInteractivity::PasteTextFromClipboard()
+    {
+        // attach ControlInteractivity::_sendPastedTextToConnection() as the
+        // clipboardDataHandler. This is called when the clipboard data is
+        // loaded.
+        auto clipboardDataHandler = std::bind(&ControlInteractivity::_sendPastedTextToConnection, this, std::placeholders::_1);
+        auto pasteArgs = winrt::make_self<PasteFromClipboardEventArgs>(clipboardDataHandler);
+
+        // send paste event up to TermApp
+        _PasteFromClipboardHandlers(*this, *pasteArgs);
+    }
+
+    // Method Description:
+    // - Pre-process text pasted (presumably from the clipboard)
+    //   before sending it over the terminal's connection.
+    void ControlInteractivity::_sendPastedTextToConnection(const std::wstring& wstr)
+    {
+        _core->PasteText(winrt::hstring{ wstr });
+    }
+
+    void ControlInteractivity::PointerPressed(const winrt::Windows::Foundation::Point mouseCursorPosition,
+                                              TerminalInput::MouseButtonState buttonState,
+                                              const unsigned int pointerUpdateKind,
+                                              const uint64_t timestamp,
+                                              const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                              const bool /*focused*/,

```

Comment:
hmm, interactivity doesn't care about focus? why have this parameter in our new clean interface then?

Reply:
yea that's vestigial. I thought all the mouse-like handlers would need the same args. They do _not_.